src/Migration.php

```diff
@@ -88,26 +104,34 @@ class Migration extends Expression
      *
      * @return Migration Subclass
      */
-    public static function getMigration($source, $params = []): self
+    public static function of($source, $params = []): self
     {
-        $c = static::getConnection($source);
-
-        switch ($c->driver) {
-            case 'sqlite':
-                return new Migration\SQLite($source, $params);
-            case 'mysql':
-                return new Migration\MySQL($source, $params);
-            case 'pgsql':
-                return new Migration\PgSQL($source, $params);
-            case 'oci':
-                return new Migration\Oracle($source, $params);
-            default:
-                throw new Exception([
-                    'Not sure which migration class to use for your DSN',
-                    'driver' => $c->driver,
-                    'source' => $source,
-                ]);
+        $connection = static::getConnection($source);
+
+        $migrator = self::$registry[$connection->driver] ?? static::class;
+
+        // if used within a subclass Migration method will create migrator of that class
+        // if $migrator class is the generic class Migration then migrator was not resolved correctly
+        if ($migrator == __CLASS__) {
+            throw new Exception([
+                'Not sure which migration class to use for your DSN',
+                'driver' => $connection->driver,
+                'source' => $source,
+            ]);
         }
+
+        return new $migrator($source, $params);
+    }
+
+    public static function register($driver, $migrator = null)
+    {
+        if (is_array($drivers = $driver)) {
+            foreach ($drivers as $driver => $migrator) {
+                self::register($driver, $migrator);
```

Comment:
Maybe use `static` instead of `self`. Not 100% sure, but I think that's better in case you extend this class.

Reply:
Based on above explanation `register` method will only work with `Migration` class only.

Reply:
Also maybe you can add comment in code to explain in short why `self` is needed there not `static`?
Or at least something like _Self here should be used instead of static. Do not change!!!_ :)